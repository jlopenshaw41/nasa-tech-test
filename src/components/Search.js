import React, { useState } from "react";
import "../styles/Search.css";

const Search = () => {
  const [value, setValue] = useState("");
  return (
    <>
      <input className="search-input" type="text" data-testid="search-input" onChange={(e) => setValue(e.target.value)}></input>
    </>
  );
};

export default Search;
