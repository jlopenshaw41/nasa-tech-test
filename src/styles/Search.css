.search-form {
  display: flex;
  flex-direction: row;
}

.search-input {
  border: 1px solid #807e7e;
  border-right: none;
  border-radius: 25px 0px 0px 25px;
  background-color: rgba(255,255,255,0.9);
  color: black;
  font-family: "Montserrat", sans-serif;
  font-size: 1.4rem;
  padding: 5px 5px 5px 10px;
  outline: none;
  }

  .search-btn {
    background-color: rgba(255,255,255,0.9);
    border: 1px solid #807e7e;
    border-left: none;
    border-radius: 0px 25px 25px 0px;
    color: #606060;
    font-size: 1.1rem;
    padding-left: 10px;
    padding-right: 10px;
    position: relative;
    outline: none;
  }

  .search-btn:active {
    color: black;
  }