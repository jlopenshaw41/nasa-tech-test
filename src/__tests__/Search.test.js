import { render } from "@testing-library/react";
import Search from "../components/Search";

describe("<Search />", () => {
  it("renders and matches snapshot", () => {
    const { asFragment } = render(<Search />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders as expected", () => {
    const { getByTestId } = render(<Search />);
    const input = getByTestId("search-input");
    expect(input).toBeTruthy();
  });
});
