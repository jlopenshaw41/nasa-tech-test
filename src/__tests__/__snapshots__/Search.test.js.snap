// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> renders and matches snapshot 1`] = `
<DocumentFragment>
  <input
    class="search-input"
    data-testid="search-input"
    type="text"
  />
</DocumentFragment>
`;
